### PR TITLE
SHAD-411 feat/download styles from backend

### PR DESCRIPTION
**Pull request by saulmm**
[SHAD-411](https://vilua.atlassian.net/browse/SHAD-411)
[_Related_](https://github.com/SignificoHealth/recco-android-sdk/pull/60)

This PR brings support to receive a style set in backoffice. When no style set fallbacks to the regular app logic to select/chose an app style.

image

## How to test

1. Wether the regular backend response, or mimicking via a proxy. Set a payload as follows

```json
{
  "id": "235753246562487294",
  "clientUserId": "99B3BCF690E653A177C602DD9999093B9EB29E50A3AF9A059AF3FCBFAB476A16",
  "isOnboardingQuestionnaireCompleted": false,
  "appStyle": {
    "darkColors": {
      "primary": "#ffddbeFF",
      "onPrimary": "#383b45FF",
      "background": "#383b45FF",
      "onBackground": "#ffffffFF",
      "accent": "#3ba17aFF",
      "onAccent": "#ffe5aeFF",
      "illustration": "#ffc188FF",
      "illustrationOutline": "#926500FF"
    },
    "lightColors": {
      "primary": "#2c956dFF",
      "onPrimary": "#ffffffFF",
      "background": "#fcfcfcFF",
      "onBackground": "#383b45FF",
      "accent": "#ea8822FF",
      "onAccent": "#2c2783FF",
      "illustration": "#ffc188FF",
      "illustrationOutline": "#306d49FF"
    },
    "iosFont": "avenir_next",
    "androidFont": "poppins"
  }
}
```

2. Ensure colors match the expected specified colors. Play around with fonts too.
3. Use a payload without theme, and check that the previous theme selection logic is preserved.

```json
{
  "id": "235753246562487294",
  "clientUserId": "99B3BCF690E653A177C602DD9999093B9EB29E50A3AF9A059AF3FCBFAB476A16",
  "isOnboardingQuestionnaireCompleted": false
}
```

### CODE REVIEW

File: Sources/ReccoUI/Onboarding/Splash/SplashViewModel.swift [hunk 1]
```diff
@@ -26,9 +27,18 @@ final class SplashViewModel: ObservableObject {
...
 
     private func bind() {
-        meRepository
+        cancellable = meRepository
             .currentUser
             .receive(on: DispatchQueue.main)
-            .assign(to: &$user)
+            .sink { [weak self] newUser in
+                self?.user = newUser
+                self?.maybeChangeToBackOfficeStyle()
```
[the remark this answers is not in the document]
Comment (arubio) [reply Sources/ReccoUI/Onboarding/Splash/SplashViewModel.swift:34]: This is the way to do it, yeah. Maybe I'd use unowned, as we know the SplashViewModel should not be out of memory when the closure inside that sink is called.

File: Sources/ReccoHeadless/Api/Generated/APIHelper.swift [hunk 2]
```diff
@@ -62,7 +62,7 @@ internal struct APIHelper {
                 .compactMap { value in convertAnyToString(value) }
                 .joined(separator: ",")
         }
-        return convertAnyToString(source) ?? source
+        return source
```
Comment (arubio) [inline Sources/ReccoHeadless/Api/Generated/APIHelper.swift:65]: this change here might break enums for our generated OpenApi client. Which generator version did you use?

Comment (arubio) [reply Sources/ReccoHeadless/Api/Generated/APIHelper.swift:65]: Thanks, after this change we should probably test if the SDK can retrieve sections after the initial questionnaire/onboarding. If not, we should revert this snippet back (it was a bug on the generator side)

Comment (arubio) [reply Sources/ReccoHeadless/Api/Generated/APIHelper.swift:65]: yeah, revert this change and check again. They did not fix that bug even on this last major release, maybe I should create a PR on their repo...

Comment (arubio) [reply Sources/ReccoHeadless/Api/Generated/APIHelper.swift]: yeah, but this file is also autogenerated and only affects this API. For a more general solution, modifying the helper OpenAPI uses is best I think. For now we leave it like this and think about that bash patch on another day. Maybe @robertofrontado makes quick work of it.

**Review by arubio — APPROVED** [review]

File: Sources/ReccoHeadless/Api/Generated/APIHelper.swift (hunk 2 above)
Comment (robertofrontado) [reply Sources/ReccoHeadless/Api/Generated/APIHelper.swift]: I'll add that patch to the bash script

Comment (robertofrontado) [reply Sources/ReccoHeadless/Api/Generated/APIHelper.swift]: Looking at it, its a bit strange that its not working 

```swift
internal static func convertAnyToString(_ value: Any?) -> String? {
    guard let value = value else { return nil }
    if let value = value as? any RawRepresentable {
        return "\(value.rawValue)"
    } else {
        return "\(value)"
    }
}

internal static func mapValueToPathItem(_ source: Any) -> Any {
    if let collection = source as? [Any?] {
        return collection
            .compactMap { value in convertAnyToString(value) }
            .joined(separator: ",")
    }
    return convertAnyToString(source) ?? source
}
```

Comment (arubio) [reply Sources/ReccoHeadless/Api/Generated/APIHelper.swift]: `convertAnyToString(source) ??` that code is mine and is the reason why it works, and it is what we should put on the bash patch @robertofrontado

**Pull request merged**